airflow/models/baseoperator.py

```diff
@@ -1336,6 +1336,9 @@ def get_extra_links(self, dttm: datetime, link_name: str) -> Optional[Dict[str,
     def get_serialized_fields(cls):
         """Stringified DAGs and operators contain exactly these fields."""
         if not cls.__serialized_fields:
+            from airflow.models.dag import DagContext
+
+            DagContext.push_context_managed_dag(None)
```

Comment:
@houqp can you tell more about why we need it?

Reply:
@turbaszek added more comment in my commit. you should be able to reproduce the crash with the added test case as well. let me know if you would like me to add more clarification.

Reply:
Do we need to do this for the following places too?

```
airflow/sensors/external_task_sensor.py:306:    def get_serialized_fields(cls):
```

Reply:
ok, maybe it is not required, since we don't add any dummy task.

Although I am pretty sure we were able to use Context Manager with DAG Serialization.

Reply:
looks like it's calling `super().get_serialized_fields()`, so we should be good for ExternalTaskSensor

Reply:
oh yeah realized after it and added another comment :) 

>ok, maybe it is not required, since we don't add any dummy task.

But I am still a bit surprised that we didn't catch or found this error before as I am pretty sure I have used Serialized DAG with `DAG context Manager`

Reply:
Depends on how you use the context manager. `with DAG():` works as expected, but we have huge DAGs in production that would be a lot easier to maintain without that extra indentation introduced with the `with` block. So we are using `DagContext.push_context_managed_dag(dag)` directly in the DAG definition. This is what triggered the crash.

Reply:
Got it, yeah :)